### PR TITLE
Allow event filtering based on tracing policy

### PR DESCRIPTION
**Pull request by ioandr**
This PR extends the list of filters by implementing the policy name filter. This allows Tetragon to filter events based on the name of a tracing policy.

~It also improves Tetragon's docs~ (see https://github.com/cilium/tetragon/pull/1965) and extends test suites as needed.

Closes https://github.com/cilium/tetragon/issues/1855

```release-note
add filter event based on tracing policy name
```

### CODE REVIEW

**Comment by kkourt** [comment]
Thanks! started the tests, and added a release note!

File: api/v1/tetragon/events.proto
Comment (mtardy) [inline api/v1/tetragon/events.proto:55]: not sure the URL is necessary here
```suggestion
```

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (mtardy) [inline docs/content/en/docs/getting-started/install-k8s.md:104]: So this URL is located in the `hugo.toml` file of the doc because I think it's used elsewhere, you can update it there.

**Review by mtardy — COMMENTED** [review]
Hey!! Thanks a lot for those patches, I think they all make sense and I would like to merge them! 

This is a bit of a nit but the commits are not really related, would it be possible to split this PR into two PRs? Maybe this one for the filtering based on the tracing policy name, and another one for the docs? If this is too annoying tell me and we'll figure something else.

Also if you can rebase since we fixed something in the CI in the meantime. Sorry for the delay, it was the Christmas period! 🎄

**Comment by willfindlay** [comment]
I'm also good with merging this once @mtardy's comments have been addressed.

**Comment by jrfastab** [comment]
+1 appears useful please address feedback and we can get this merged! :)

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (kkourt) [inline docs/content/en/docs/getting-started/install-k8s.md:89]: Unfortunately, this does not seem to work for bash which is what I would guess many people use.

How about:
```
EXTRA_HELM_FLAGS="--set tetragon.hostProcPath=/procHost" # flags for helm install
helm install tetragon $(echo ${EXTRA_HELM_FLAGS}) cilium/tetragon -n kube-system
```

Maybe there is something better?

**Review by kkourt — CHANGES_REQUESTED** [review]
Thanks!

Everything looks good to me.

I'd prefer if the last patch was split into multiple patches, such as
 - documentation fixes (typos)
 - core code changes
 - tetra CLI

But I don't think we should block the PR on this.

Could you please address the comments from everyone and have the documentation change also work in bash?

Other than that, good to go from my side!

File: api/v1/tetragon/events.proto
Comment (ioandr) [reply api/v1/tetragon/events.proto:55]: ACK, will remove it.

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (ioandr) [reply docs/content/en/docs/getting-started/install-k8s.md:104]: Thanks for the pointer, I was not aware of the `hugo.toml` file. Will update it there.

Comment (ioandr) [reply docs/content/en/docs/getting-started/install-k8s.md:104]: Actually, the current value of `demo_app_url` points to version `v1.15.0-pre.1` which is more recent compared to `1.14.4`.

However, it seems that this URL is not correctly rendered, regardless its value in `hugo.toml`:

https://tetragon.io/docs/getting-started/install-k8s/#deploy-demo-application

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (ioandr) [reply docs/content/en/docs/getting-started/install-k8s.md:89]: Removing the double quotes when setting the `EXTRA_HELM_FLAGS` env var does the trick for me for both shells. Not sure why I kept them in my patch.

- **zsh**

```bash
➜  tetragon git:(feature-i-doc-fixes) ps -p $$                                                              
    PID TTY          TIME CMD
 158412 pts/1    00:00:07 zsh

➜  tetragon git:(feature-i-doc-fixes) EXTRA_HELM_FLAGS=(--set tetragon.hostProcPath=/procHost --set tetragon.foo=bar)

➜  tetragon git:(feature-i-doc-fixes) helm install tetragon ${EXTRA_HELM_FLAGS[@]} cilium/tetragon -n kube-system
NAME: tetragon
LAST DEPLOYED: Fri Jan 12 00:36:04 2024
NAMESPACE: kube-system
STATUS: deployed
REVISION: 1
TEST SUITE: None

➜  tetragon git:(feature-i-doc-fixes) helm get values tetragon -n kube-system                                      
USER-SUPPLIED VALUES:
tetragon:
  foo: bar
  hostProcPath: /procHost
```
- **bash**

```bash
gozek@nexus:~/go/src/github.com/tetragon (feature-i-doc-fixes) $ ps -p $$
    PID TTY          TIME CMD
 191941 pts/1    00:00:00 bash

gozek@nexus:~/go/src/github.com/tetragon (feature-i-doc-fixes) $ EXTRA_HELM_FLAGS=(--set tetragon.hostProcPath=/procHost --set tetragon.foo=bar)

gozek@nexus:~/go/src/github.com/tetragon (feature-i-doc-fixes) $ helm install tetragon ${EXTRA_HELM_FLAGS[@]} cilium/tetragon -n kube-system
NAME: tetragon
LAST DEPLOYED: Fri Jan 12 00:38:20 2024
NAMESPACE: kube-system
STATUS: deployed
REVISION: 1
TEST SUITE: None

gozek@nexus:~/go/src/github.com/tetragon (feature-i-doc-fixes) $ helm get values -n kube-system tetragon
USER-SUPPLIED VALUES:
tetragon:
  foo: bar
  hostProcPath: /procHost
```

An alternative is to use `setopt shwordsplit` for `zsh` only which does not split words by default. Also see:

1. https://zsh.sourceforge.io/FAQ/zshfaq03.html#l18
2. https://stackoverflow.com/questions/6715388/variable-expansion-is-different-in-zsh-from-that-in-bash

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (ioandr) [reply docs/content/en/docs/getting-started/install-k8s.md:104]: IIUC and since we are using [Hugo's shortcodes](https://gohugo.io/content-management/shortcodes/#param) we have to use parameters in `.md` files.

Maybe this commit is related? cca76556fb55810ee9a239d83e4f165d9c4fa5ae

Moving `demo_app_url` under the `[params]` section does the trick.

**Comment by ioandr** [comment]
Update:

1. I moved doc-related changes to a new PR - see https://github.com/cilium/tetragon/pull/1965
2. I broke down remaining changes introduced by the original PR into 3 commits as discussed above
3. I rebased over latest `main` and force-pushed to the branch of the original PR
4. I did not have the time to re-test functionality, I can do that in the following days (if needed)

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (ioandr) [reply docs/content/en/docs/getting-started/install-k8s.md]: We should probably move this discussion to https://github.com/cilium/tetragon/pull/1965/commits/768b350b52701862f224769b2eb367f2d65c4c6e.

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (ioandr) [reply docs/content/en/docs/getting-started/install-k8s.md]: We should probably move this discussion to https://github.com/cilium/tetragon/pull/1965.

File: docs/content/en/docs/getting-started/install-k8s.md
Comment (kkourt) [reply docs/content/en/docs/getting-started/install-k8s.md]: Thanks!

> Of course I am OK with this approach if you prefer it.

No preference as long is it works for both shells.

**Comment by ioandr** [comment]
> Is there testing that is not covered by the unit tests? As far as I can see, there were no changes in the code (other than the rebasing) and you have unit tests so, at least from my side, it's good to merge.

Indeed, just mentioned this for completeness. I'm good to proceed.

**Comment by ioandr** [comment]
@mtardy now that your comments have been addressed maybe approve this?

File: cmd/tetra/getevents/getevents.go
Comment (mtardy) [inline cmd/tetra/getevents/getevents.go:222]: only nit, maybe this could have been `policies` since it's easier to type than `policy-names` but this is mostly personal preference!

**Review by mtardy — APPROVED** [review]
Thanks a lot, and thanks for adding tests, let's merge that :)

**Comment by mtardy** [comment]
I can't merge because of this @willfindlay, could you review it?
https://github.com/cilium/tetragon/blob/0a9b71e5d6d0e1ad59cdab61ad13d9eca9732cbd/.github/CODEOWNERS#L4-L5

In the meantime @ioandr if you can ever rebase this on main (I'm sorry I wanted to merge this now), I fixed the CI issue that made vmtests fail early.

File: cmd/tetra/getevents/getevents.go
Comment (ioandr) [reply cmd/tetra/getevents/getevents.go:222]: No strong opinion from my side on this, LMK if you prefer `policies` over `policy-names`.

**Comment by mtardy** [comment]
Thanks for the rebase, everything is green, now just waiting on William's ack.

**Review by willfindlay — APPROVED** [review]

**Comment by willfindlay** [comment]
Merged, thanks

**Pull request merged**
